src/applications/vaos/containers/DateTimeSelectPage.jsx

```diff
@@ -85,8 +93,21 @@ export class DateTimeSelectPage extends React.Component {
   };
 
   goForward = () => {
+    this.props.routeToNextAppointmentPage(this.props.router, pageKey);
+  };
+
+  validate = () => {
     if (this.props.data.calendarData?.selectedDates?.length) {
-      this.props.routeToNextAppointmentPage(this.props.router, pageKey);
+      this.setState({ validationError: null });
+      this.goForward();
+    } else {
+      this.setState(
+        {
+          validationError:
+            'Please select a preferred date for your appointment',
+        },
+        () => scrollAndFocus('#vaos-calendar__validation-msg'),
```

Comment:
We've had problems with this post-setState callback in the past. Another option would be to do this in `componentDidUpdate`, when the validationError changes.